src/picosvg/svg.py

```diff
@@ -1103,25 +1148,28 @@ def topicosvg(self, *, ndigits=3, inplace=False):
 
         self._update_etree()
 
+        # Discard useless content
         self.remove_nonsvg_content(inplace=True)
         self.remove_comments(inplace=True)
         self.remove_anonymous_symbols(inplace=True)
         self.remove_title_meta_desc(inplace=True)
+
+        # Simplify things that simplify in isolation
         self.apply_style_attributes(inplace=True)
         self.resolve_nested_svgs(inplace=True)
```

Comment:
once this PR is merged, we can go on and fix that TODO at the end of `_unnest_svg` method. Now that transform and clip-paths work nicely together, we can support the clipping behavior of nested `<svg>` elements: i.e. define a new `clipPath` element containing a rectangle the size of the nested svg viewport and set clip-path attribute on the `<g>` element that we swap in for the nested `<svg>`.

Reply:
I think this is all you need to implement that

```diff
diff --git a/src/picosvg/svg.py b/src/picosvg/svg.py
index 0e45249..749bbca 100644
--- a/src/picosvg/svg.py
+++ b/src/picosvg/svg.py
@@ -921,9 +921,13 @@ class SVG:
         if transform != Affine2D.identity():
             g.attrib["transform"] = transform.tostring()

-        # TODO Define a viewport-sized clipPath once transform+clip-path issue is fixed
-        # https://github.com/googlefonts/picosvg/issues/200
-        return (g,)
+        clip_path = etree.Element(
+            f"{{{svgns()}}}clipPath", {"id": self._new_id("nested-svg-viewport-%d")}
+        )
+        clip_path.append(to_element(SVGRect(width=width, height=height)))
+        g.attrib["clip-path"] = f"url(#{clip_path.attrib['id']})"
+
+        return (clip_path, g)

     def resolve_nested_svgs(self, inplace=False):
         """Replace nested <svg> elements with equivalent <g> with a transform.
```

Reply:
FYI, I updated the code above for clipping nested svg; it now supports optional overflow="visible"

Reply:
Included this, fixed the flag svg, discovered a new bug where we would leave behind pointless stroke attributes (stroke-width present but no stroke value), fixed that too. Sick.